samples/matter/common/src/system_event.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <cstdint>
+
+#include "event_manager.h"
+#include "led_widget.h"
+
+enum class SystemEventType : uint8_t {
+	None = 0,
+	Button,
+	ButtonPushed,
+	ButtonReleased,
+	Timer,
+	UpdateLedState,
+	IdentifyStart,
+	IdentifyStop,
+	SoftwareUpdate,
+	FactoryReset,
+	AdvertisingStart
+};
+
+struct SystemEvent : public Event {
+	SystemEvent()
+		: Event(EventSource::System, static_cast<uint8_t>(SystemEventType::None), nullptr, this, sizeof(*this))
```

Comment:
There seems to be a lot of casing of the EventTypes into numeric types, maybe it's better to define them as non-strongly typed enums?